### ISSUE TITLE
PlasmaPhase electron enthalpy depends on gas temperature

### ISSUE THREAD

**Issue by speth** [Cantera/cantera#1306]
**Problem description**

The enthalpy of the electron species in `PlasmaPhase` is calculated incorrectly, such that the enthalpy of the electron depends on both the electron temperature and the temperature of the non-electron species. This appears to be a result of calculating the non-dimensional enthalpy using the electron temperature in the overridden `updateThermo` method, but then dimensionalizing it using the gas temperature in `IdealGasPhase::getPartialMolarEnthalpies` (which is not overridden).

**Steps to reproduce**

```python
>>>[1]: import cantera as ct
>>>[2]: p = ct.Solution('test/data/oxygen-plasma.yaml',
                        'discretized-electron-energy-plasma', transport_model='None')
>>>[3]: p.TPX = 300, 101325, 'E: 1'
>>>[4]: p.Te
379740.6337340625
>>>[5]: p.enthalpy_mole
6230904.132044763
>>>[6]: p.TP = 301, None
>>>[7]: p.enthalpy_mole
6251673.8124849135
>>>[8]: p.Te
379740.6337340625
```

**Behavior**

The molar enthalpy for a phase of pure electrons should only depend on `Te`, not `T`.

**System information**

- Cantera version: 2.6.0 or `main` at 8fbb4bcd9

**Attachments**

**Additional context**

This was discovered as part of https://github.com/Cantera/cantera/pull/1299, which implements https://github.com/Cantera/enhancements/issues/114.

**Comment by BangShiuh** [comment]
I think some system thermodynamic parameter might not be applicable for plasma. Ex. G = H - T S. But for convenience, it might be good to have a option to neglect the contribution of electron so that the system thermodynamics can still be obtained.

**Comment by speth** [comment]
Yes, you're right, some of the bulk properties will not satisfy those relationships when the electron temperature is different from the gas species, but I think at least g_electron = h_electron - T_electron * s_electron should still be true.

**Comment by ischoegl** [comment]
Fixed by #1323
